### DISCUSSION TITLE
Welcome to JReleaser Discussions!

### DISCUSSION THREAD

**Discussion by aalmiray** [jreleaser/jreleaser#8] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by timoreichen** [comment]
I see great value in jrelease. Though as javafx developer for endusers, what i am missing is a tool to release and autoupdate signed/notarized apps. That said for windows and macos. For now i did not find any solution to easy release updates of notarized apps. Are there any plans supporting that? Or is it possible already with any of the supported frameworks?

**Reply by aalmiray** [reply]
I think it depends on the process you currently have in place to sign and/or notarize a JavaFX application to be distribution by a particular distribution channel. JReleaser could be configured to automate said process but first we have to know what is, how it works (expected inputs/outputs), and so on.

**Comment by timoreichen** [comment]
In short for my application it builds a fat jar with maven. Then a script is launched to run jpackager to build the app-file which then gets signed. Then for macos the dmg has to be signed as well. In the end i have a signed/notarized dmg. Its not ment to be put on an app-store but to fulfill the hurdle of trust on the user side. No one will probably install an application when security warnings are popping up. So for now with every release the user has to download a dmg file to reinstall. I am looking for a simple solution to make that easy. 

Which is the main painpoint building desktop applications with javafx - in my opinion. There are very good frameworks to write it once but none to autoupdate the most simple application. For Endusers ;-) which unfortunately won't use brew etc. Maybe i am wrong.

**Comment by aalmiray** [comment]
There are options to update a JavaFX application:

- https://github.com/update4j/update4j
- https://github.com/moditect/layrry

Layrry requires your application to be fully modular. The problem with self-updating applications is that they can easily break the trust circle by downloading additional JARs. Hence why updating applications as a whole is the way to go.
